src/go/configgenerator/filterconfig/filter_generator_test.go

```diff
@@ -0,0 +1,954 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterconfig
+
+import (
```

Comment:
What is the plan for this test file in g3? Will any of these fail in g3?

 Just like the filter generator files, it will be easier if we can move each filter-specific test to its own file. Then we copy just the files we need.

Reply:
ACK

Was planning to add PerRouteConfig tests for individual filter_gen along with sort out tests in this test into specific one. Will do this PR.

Reply:
Move the tests with PerRouteConfig into their own test files for better readability.